plugin.video.southpark_unofficial/southpark.py

```diff
@@ -0,0 +1,596 @@
+		except:
+			pass
+	return string
+
+def _encode(string):
+	try:
+		return string.encode('UTF-8','replace')
+	except UnicodeError:
+		return string
+
+def _translation(addon, id):
+	return _encode(addon.getLocalizedString(id))
+
+def _premier_timeout(premiere):
+	try:
+		diff = int(premiere) - int(time.time())
+	except ValueError:
+		date = time.strptime(premiere, "%d.%m.%Y") # 30.09.2015
+		diff = int(time.mktime(date)) - int(time.time())
+		
+	if diff < 0:
+		diff = 0;
+	days = int(diff / 86400)
+	hours = int((diff % 86400) / 3600)
+	mins =  int((diff % 3600) / 60)
+	return "%02dd %02dh %02dm".format(days, hours, mins)
+
+def log_debug(message):
+	xbmc.log("[sp.addon] {0}".format(message), xbmc.LOGDEBUG)
+
+def log_error(message):
+	xbmc.log("[sp.addon] {0}".format(message), xbmc.LOGERROR)
+
+def _http_get(url):
+	log_debug("http get: {0}".format(url))
+	req = urllib2.Request(url)
+	req.add_header('User-Agent', USER_AGENT)
+	response = urllib2.urlopen(req)
+	link = response.read()
+	response.close()
+	return link
+
+def _parameters_string_to_dict(parameters):
+	paramDict = {}
+	if parameters:
+		paramPairs = parameters[1:].split("&")
+		for paramsPair in paramPairs:
+			paramSplits = paramsPair.split('=')
+			if (len(paramSplits)) == 2:
+				paramDict[paramSplits[0]] = paramSplits[1]
+	return paramDict
+
+def _save_subs(fname, stream):
+	data = _unescape(_http_get(stream))
+	output = open(fname,'wb')
+	output.truncate()
+	output.write(data)
+	output.close()
+	log_debug("Saved subtitle {0}.".format(fname))
+	return stream
+
+class SP_Paths(object):
+	"""South Park plugin Paths"""
+	def __init__(self, addon_id):
+		super(SP_Paths, self).__init__()
+		self.PLUGIN_ICON      = xbmc.translatePath('special://home/addons/{0}/icon.png'.format(addon_id))
+		self.DEFAULT_FANART   = xbmc.translatePath('special://home/addons/{0}/fanart.jpg'.format(addon_id))
+		self.DEFAULT_IMGDIR   = xbmc.translatePath('special://home/addons/{0}/imgs/'.format(addon_id))
+		self.TEMPORARY_FOLDER = xbmc.translatePath('special://temp/southpark')
+		if not os.path.isdir(self.TEMPORARY_FOLDER):
+			os.mkdir(self.TEMPORARY_FOLDER, 0o755)
+
+class KodiParams(object):
+	"""docstring for KodiParams"""
+	def __init__(self, param_string):
+		super(KodiParams, self).__init__()
+		params = _parameters_string_to_dict(param_string)
+		self.PARAM_MODE         = urllib.unquote_plus(params.get('mode'     , ''))
+		self.PARAM_URL          = urllib.unquote_plus(params.get('url'      , ''))
+		self.PARAM_EP_TITLE     = urllib.unquote_plus(params.get('title'    , ''))
+		self.PARAM_EP_THUMBNAIL = urllib.unquote_plus(params.get('thumbnail', ''))
+
+	def debug(self):
+		log_debug("PARAM_MODE:         {0}".format(self.PARAM_MODE        ))
+		log_debug("PARAM_URL:          {0}".format(self.PARAM_URL         ))
+		log_debug("PARAM_EP_TITLE:     {0}".format(self.PARAM_EP_TITLE    ))
+		log_debug("PARAM_EP_THUMBNAIL: {0}".format(self.PARAM_EP_THUMBNAIL))
```

Comment:
reduce the log spamming, by at least combining those into one log line

Reply:
i've commented the line 571 & 572, so they will never be called.